api-tests/core/content-manager/find-relations.test.api.js

```diff
@@ -0,0 +1,803 @@
+
+    await modelsUtils.modifyComponent(compo(true));
+
+    strapi = await createStrapiInstance();
+    rq = await createAuthRequest({ strapi });
+
+    await rq({
+      method: 'POST',
+      url: '/i18n/locales',
+      body: {
+        code: extraLocale,
+        name: `French (${extraLocale})`,
+        isDefault: false,
+      },
+    });
+
+    // Create draft products
+    const [skate, chair, candle, table, porte, fenetre] = await Promise.all([
+      createEntry(productUid, { name: 'Skate' }),
+      createEntry(productUid, { name: 'Chair' }),
+      createEntry(productUid, { name: 'Candle' }),
+      createEntry(productUid, { name: 'Table' }),
+      // We create products in French in order to test that we can cant find
+      // aviailable relations in a different locale
+      createEntry(productUid, { name: 'Porte', locale: extraLocale }),
+      createEntry(productUid, { name: 'Fenêtre', locale: extraLocale }),
+    ]);
+    data.products.draft.push(
+      skate.data,
+      chair.data,
+      candle.data,
+      table.data,
+      porte.data,
+      fenetre.data
+    );
+    const productMapping = {
+      skate: data.products.draft[0],
+      chair: data.products.draft[1],
+      candle: data.products.draft[2],
+      table: data.products.draft[3],
+      porte: data.products.draft[4],
+      fenetre: data.products.draft[5],
+    };
+
+    // Publish Skate and Chair
+    const [publishedSkate, publishedChair] = await Promise.all([
+      rq({
+        url: `/content-manager/collection-types/${productUid}/${productMapping.skate.id}/actions/publish`,
+        method: 'POST',
+      }),
+      rq({
+        url: `/content-manager/collection-types/${productUid}/${productMapping.chair.id}/actions/publish`,
+        method: 'POST',
+      }),
+    ]);
+    data.products.published.push(publishedSkate.body.data, publishedChair.body.data);
+
+    /**
+     * Transform the draft relations into published relations if they exist
+     */
+    const buildPublishedRelations = (draftRelations, publishedProducts) => {
+      const publishedRelations = JSON.parse(JSON.stringify(draftRelations));
+
+      const replaceWithPublished = (draftProduct) => {
+        const published = publishedProducts.find(
+          (publishedProduct) => publishedProduct.id === draftProduct.id
+        );
+
+        return published || undefined;
+      };
+
+      // Iterate over relations and replace products with their published counterparts
+      for (const key in publishedRelations) {
+        if (Array.isArray(publishedRelations[key])) {
+          publishedRelations[key] = publishedRelations[key]
+            .map(replaceWithPublished)
+            .filter(Boolean);
+        } else if (
+          typeof publishedRelations[key] === 'object' &&
+          publishedRelations[key] !== null
+        ) {
+          publishedRelations[key] = replaceWithPublished(publishedRelations[key]);
+        }
+      }
+
+      return publishedRelations;
+    };
```

Comment:
Yeah this is a bit fiddly but the published shops only contain the counts for each relation and no details about the actual relation which we need for comparisons in the test cases. Do you think theres a better way?

Reply:
you might be able to use strapi.documents(uid).publish instead of the api, that + sending a populate object on the relations might return you what you need